model/src/main/proto/question.proto

```diff
@@ -81,3 +81,17 @@ message AnsweredQuestionOutcome {
   // Whether the answer the learner submitted is the correct answer.
   bool is_correct_answer = 2;
 }
+
+// The final score and mastery calculations of the user after completing a training session. This is an ephemeral data
+// structure that is only relevant to a particular training session.
+message UserAssessmentPerformance {
+  // Stores the user's total session score as a percentage.
+  double total_percentage_score = 1;
+
+  // Stores the user's score per skill for the training session via a map of skill IDs to percentage scores.
+  map<string, double> percentage_score_per_skill_mapping = 2;
+
+  // Stores the user's mastery degree per skill for the training session via a map of skill IDs to mastery degrees.
+  // A mastery degree is a numerical score representing the user's strength in a particular skill.
+  map<string, double> mastery_per_skill_mapping = 3;
```

Comment:
I am wondering if its worth clubbing these two into a single map. I am not so sure if it will give us a lot of benefit though, what do you think @TheRealJessicaLi ?

Reply:
I feel like since it's only 2 things, it doesn't really make a difference if we put them together - these are 2 separate maps because the score and mastery calculations occur separately. If we wanted to group them together, we would need an extra data structure and we would need to process both sets again after they're individually computed. I think this makes sense, but let me know if you think there's a better approach!

Reply:
Makes sense, your approach LGTM, thanks!